### PR TITLE
Build:Tests: Fix custom build tests, verify on Travis 

### PR DESCRIPTION
**Pull request by mgol**
### Summary ###

This PR fixes unit tests for the following builds:
1. The no-deprecated build: `custom:-deprecated`
1. The current slim build: `custom:-ajax,-effects`
1. The future (#4553) slim build: `custom:-ajax,-callbacks,-deferred,-effects`

It also adds separate Karma runs for the no-deprecated & current slim builds. Once the new slim build configuration lands, we'll have to modify what we test on Travis as well.

This PR can also be cherry-picked to `3.x-stable`.

### Checklist ###

* [x] All authors have signed the CLA at https://cla.js.foundation/jquery/jquery
* [x] New tests have been added to show the fix or feature works
* [x] Grunt build and unit tests pass locally with these changes
* ~~If needed, a docs issue/PR was created at https://github.com/jquery/api.jquery.com~~

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -74,7 +74,9 @@
     "test:browser": "grunt && grunt karma:main",
     "test:esmodules": "grunt && grunt karma:esmodules",
     "test:amd": "grunt && grunt karma:amd",
-    "test": "grunt && grunt test:slow && grunt karma:main && grunt karma:esmodules && grunt karma:amd",
+    "test:no-deprecated": "grunt test:prepare && grunt custom:-deprecated && grunt karma:main",
+    "test:slim": "grunt test:prepare && grunt custom:-ajax,-effects && grunt karma:main",
```
Comment (timmywil) [inline package.json]: You probably already planned to do this, but change to `custom:slim` when that's merged.

Comment (mgol) [reply package.json]: @timmywil PR updated, I rebased it & added a new commit modifying this line.

**Comment by mgol** [comment]
Changing milestones, I'll submit a separate PR for the `3.x-stable` branch as the changes there might be slightly different.

**Comment by mgol** [comment]
The PR for `3.x-stable` is submitted now: #4595.

**Comment by mgol** [comment]
Ah, no, it's not, wrong PR, sorry.

**Pull request merged**